### PR TITLE
VCST-5089: Configure per-store Asset Public URL for XAPI asset links

### PR DESCRIPTION
**Pull request by OlegoO**
## Description
feat: Configure per-store Asset Public URL for XAPI asset links

image

### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-5089
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.Store_3.1006.0-pr-171-e5e1.zip

---

> [!NOTE]
> **Medium Risk**
> Requires DB migrations and changes how public asset URLs are built for storefronts when configured; misconfigured `AssetPublicUrl` or `KnownAssetHosts` could break image links until corrected.
> 
> **Overview**
> Adds **per-store asset public URL** so Experience API consumers can rewrite image/asset links onto a store-specific CDN or domain instead of the global platform URL.
> 
> **Data & API surface:** New nullable `Store.AssetPublicUrl` on the domain model and `Store` table (migrations for SQL Server, MySQL, PostgreSQL), mapped through `StoreEntity` for CRUD/export-import.
> 
> **Resolution:** Introduces `IStoreAssetPublicUrlResolver` / `StoreAssetPublicUrlResolver` — when `AssetPublicUrl` is set, relative paths are joined to the base URL and absolute http(s) URLs are rebased (path, query, fragment preserved); `data:`, `blob:`, and protocol-relative URLs are left unchanged. Optional `VirtoCommerce:StoreAssets:KnownAssetHosts` limits which source hosts get rebased; empty list means rebase any absolute URL.
> 
> **Admin UI:** Store detail blade gains a **URLs** section with storefront URL and **Store asset URL** (with hint). Stores list adds state filter panel, shows URL under name, and context-menu actions to copy store id/URL and open storefront in a new tab. Locales updated across languages.
> 
> **Docs & tests:** README documents configuration and the new service; broad unit tests cover resolver edge cases. Minor EF/tooling package bumps on data provider projects.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e5e14f9aca8a40c3343145a6bdcdabf80295fb5f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/VirtoCommerce.StoreModule.Data/Services/StoreAssetPublicUrlResolver.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+        if (IsAbsolute(url, out var uri))
+        {
+            // When known hosts are configured, replace only listed hosts; other hosts are external.
+            if (_knownAssetHosts.Count > 0 && !_knownAssetHosts.Contains(uri.Host))
+            {
+                return url;
+            }
+
+            return RebaseHost(assetPublicUrl, url);
```
Comment (DmitryGrishinVirtoworks) [inline src/VirtoCommerce.StoreModule.Data/Services/StoreAssetPublicUrlResolver.cs:69]: IsAbsolute only accepts http/https schemes, so a protocol-relative URL like //cdn.example.com/img.jpg isn't recognized as absolute and falls through to CombineUrl, producing a malformed result (https://cdn.store1.com//cdn.example.com/img.jpg). Is this input format expected to occur in stored asset URLs?

I didn't find the way to do that, but just in case

File: src/VirtoCommerce.StoreModule.Web/Scripts/blades/store-detail.tpl.html [hunk 2]
```diff
@@ -44,6 +44,25 @@
...
+                <div class="form-group">
+                    <label class="form-label">{{ 'stores.blades.store-advanced.labels.store-url' | translate }}</label>
+                    <div class="form-input">
+                        <input ng-model="blade.currentEntity.url" type="url" placeholder="{{ 'stores.blades.store-advanced.placeholders.store-url' | translate }}">
+                    </div>
+                </div>
+                <div class="form-group">
+                    <label class="form-label">{{ 'stores.blades.store-advanced.labels.store-asset-url' | translate }} <i class="form-ico fa fa-question-circle __link __lightblue" ng-click="blade.assetPublicUrl_descrVisible=!blade.assetPublicUrl_descrVisible"></i></label>
+                    <div class="form-input">
+                        <input ng-model="blade.currentEntity.assetPublicUrl" type="url" placeholder="{{ 'stores.blades.store-advanced.placeholders.store-asset-url' | translate }}">
```
[the remark this answers is not in the document]
Comment (OlegoO) [reply src/VirtoCommerce.StoreModule.Web/Scripts/blades/store-detail.tpl.html:59]: Fixed
